crates/workspace/src/workspace.rs

```diff
@@ -6829,6 +6829,22 @@ pub fn open_new(
     })
 }
 
+pub fn with_workspace<F>(cx: &mut App, f: F)
+where
+    F: FnOnce(&mut Workspace, &mut Window, &mut Context<Workspace>) + 'static + Send,
+{
+    if let Some(app_state) = AppState::try_global(cx).and_then(|weak| weak.upgrade()) {
+        if let Some(window) = cx.active_window() {
+            if let Some(workspace_window) = window.downcast::<Workspace>() {
+                let _ = workspace_window.update(cx, f);
```

Comment:
I think this is wrong, as we're not allowed to skip the error without logging.

```suggestion
                workspace_window.update(cx, f).context("updating existing workspace").log_err();
```

And here seem to lie the culprit of our issues with the global-only workspace action handling, as when I `dbg!` this result, I see:

```
[crates/workspace/src/workspace.rs:6840:25] workspace_window.update(cx, f) = Err(
    window not found,
)
```

This is odd, as we're literally having the workspace on our hands, so I wonder what are we doing wrong?